server/enterprise/message_export/shared/shared.go

```diff
@@ -398,6 +368,19 @@ func GetJoinsAndLeavesForChannel(startTime int64, endTime int64, channelMembersH
 	return joins, leaves
 }
 
+func GetPostAttachments(db MessageExportStore, post *model.MessageExport) ([]*model.FileInfo, error) {
+	// if the post included any files, we need to add special elements to the export.
+	if len(post.PostFileIds) == 0 {
```

Comment:
Checking that `post` is guaranteed to be non-nil.

Reply:
Yep, as per last PR's check: https://github.com/mattermost/mattermost/pull/29191#discussion_r1842841661